pycontrails/datalib/ecmwf/model_level_era5.py

```diff
@@ -0,0 +1,501 @@
+"""Model-level ERA5 data access.
+
+This module supports
+
+- Retrieving model-level ERA5 data by submitting MARS requests through the Copernicus CDS.
+- Processing retrieved GRIB files to produce netCDF files on target pressure levels.
+- Local caching of processed netCDF files.
+- Opening processed and cached files as a :class:`pycontrails.MetDataset` object.
+
+Consider using :class:`pycontrails.datalib.ecmwf.ARCOERA5`
+to access model-level data from the nominal ERA5 reanalysis between 1959 and 2022.
+:class:`pycontrails.datalib.ecmwf.ARCOERA5` accesses data through Google's
+`Analysis-Ready, Cloud Optimized ERA5 dataset <https://cloud.google.com/storage/docs/public-datasets/era5>`_
+and has lower latency than this module, which retrieves data from the
+`Copernicus Climate Data Store <https://cds.climate.copernicus.eu/#!/home>`_.
+This module must be used to retrieve model-level data from ERA5 ensemble members
+or for more recent dates.
+
+This module requires the following additional dependency:
+
+- `metview (binaries and python bindings) <https://metview.readthedocs.io/en/latest/python.html>`_
+"""
+
+from __future__ import annotations
+
+import collections
+import contextlib
+import hashlib
+import logging
+import os
+import warnings
+from datetime import datetime
+from typing import Any
+
+from overrides import overrides
+
+LOG = logging.getLogger(__name__)
+
+import pandas as pd
+import xarray as xr
+
+import pycontrails
+from pycontrails.core import cache, datalib, met_var
+from pycontrails.core.met import MetDataset, MetVariable
+from pycontrails.datalib.ecmwf import variables as ecmwf_var
+from pycontrails.datalib.ecmwf.common import ECMWFAPI
+from pycontrails.datalib.ecmwf.model_levels import pressure_levels_at_model_levels
+from pycontrails.utils import dependencies, temp
+
+MODEL_LEVEL_VARIABLES = [
```

Comment:
Is there a reason these are different than the pressure level variables? I would expect thes variables to be the same as in the other `ERA5` class

Reply:
Yes! It's because not all ERA5 pressure-level variables are available on model levels. (Model levels aren't missing anything we care about, just RH and PV).